src/binary/non_blocking/binary_buffer.rs

```diff
@@ -56,7 +56,7 @@ impl<A: AsRef<[u8]>> BinaryBuffer<A> {
             data: &self.data,
             start: self.start,
             end: self.end,
-            total_consumed: 0,
+            total_consumed: self.total_consumed,
```

Comment:
🗺️ Previously, the transaction buffer would track its own total consumption. However, I realized this wasn't terribly useful in its existing role. Having it copy the original buffer's `total_consumed` means that if it encounters errors, the error position that it reports will be the absolute offset in the Ion stream which is what we want.

Incidentally, this piece of information was the only reason we stored a reference to the original buffer in the `TxReader`, so that's been removed as well.